op-chain-ops/genesis/config.go

```diff
@@ -289,6 +289,9 @@ type DeployConfig struct {
 
 	// UseInterop is a flag that indicates if the system is using interop
 	UseInterop bool `json:"useInterop,omitempty"`
+
+	// DeployCeloContracts indicates whether to deploy Celo contracts.
```

Comment:
```suggestion
	// DeployCeloContracts indicates whether to deploy Celo contracts, this should be set to true when initialising a new chain, and false in the case of a migration.
```

Reply:
Actually, do we even need this parameter, isn't this implied by the `l2GenesisBlockNumber` if it's 0 then we will deploy the contracts, if not then it must be a migration?

Reply:
I think it's nicer to be explicit and add the option, but have no strong opinion.

Reply:
If there is a valid case where `deployCeloContracts` is true and `l2GenesisBlockNumber` is greater than zero, or vi ce versa, then I guess we need it. Otherwise it seems like it's just increasing the amount of required config and opening up a path for misconfiguration which wouldn't exist otherwise.

Reply:
@carterqw2 thoughts?

Reply:
Makes sense to me. I can't think of a valid case.

Reply:
@piersy Any idea if this the `L2GenesisBlockNumber` is actually set somewhere to a non-zero value? I don't see it in any file under `packages/contracts-bedrock/deploy-config/`

Reply:
Hmm, well I'm actually I did a bit of digging and it seems in our case the `L2GenesisBlockNumber` is irrelevant.

It's passed in the config to `genesis.BuildL2Genesis` and used there as the block number of the built L2 genesis and as the `BedrockBlock` in the `ChainConfig` included with that genesis.

But then the only thing we use from that genesis is `Alloc` from that genesis at `main.go 257` and we load the chain config from the datadir that we are migrating and manually set the `BedrockBlock` to the cel2 block number at `main.go 364`.

It seems like we don't need the`L2GenesisBlockNumber` so I don't think we can make any decision based on it. So I think it's fine to keep the explicit `DeployCeloContracts` flag.